### ISSUE TITLE
add --x-forward

### ISSUE THREAD

**Issue by AwokeKnowing** [docker/cli#919]
it's becoming very standard for many workflows to include running a container like:

I'm proposing a flag --x-forward which basically does:
-e "DISPLAY" -v "/tmp/.X11-unix:/tmp/.X11-unix:rw"

I'm not saying that's the best way to implement. perhaps something like runtime=nvidia-x  which would add x in the image and map things etc.

But clearly it would massively increase user friendliness and future-proof a lot of documentation on using containers.  also it could avoid a lot of people doing xhost +  an forgetting about it.

**Comment by justyntemme** [comment]
I would love to work on this, as I use it fairly frequently. I would like some suggestions on the best implementation before i get started.

**Comment by cpuguy83** [comment]
I don't think we should do this. It's simple to setup either an alias that you can run as a subshell or some env var to put this stuff into.

Meanwhile these values aren't necessarily standard.

**Comment by AwokeKnowing** [comment]
@cpuguy83  it is exactly the 'non-standard' part which makes this a good candidate.   for example, we moved from the 'simple' nvidia-docker 'alias' which just mapped various things into various places to a 'standard' plugin.  

I'm saying docker is missing a standard way to get X from a container to showing on your screen, yet a multitude of people need to do this.

**Comment by AwokeKnowing** [comment]
@cpuguy83 I understand.  But docker revolutionized computing.  People now build and deliver software that way, including things like ROS and Gazebo,  and robotics sim stuff and outputing to X is as universal as things get in linux applications.  Certainly more so that working with a GPU.

I know I have no influence, but this will happen one way or another.   It is in fact the opposite of a 'you can just alias' type of issue.  This is the completion of docker as the right way to deliver linux applications, period.  with a well-supported way for applications to write to X and have that show on their local machines, or over ssh, docker can be the universal way to  package manage, and update linux applications, the way it practically is now for non-gui applications.  we need to decouple our gui apps from the hardware too, and X is quite universal.   As I said, people are already doing this to deliver applications, but with sh scripts to map sockets and drivers etc, and this is a great opportunity for docker  adoption.

**Comment by bsousaa** [comment]
Maybe this could be implemented through https://github.com/docker/cli/issues/3864 once accepted and implemented.

**Comment by AwokeKnowing** [comment]
> Maybe this could be implemented through #3864 once accepted and implemented.

wow, this was 5 years ago. It was a huge mistake to not have done it. it would have cemented docker as a viable generic app-delivery mechanism at a time when people we're finally not considering VM's and looking for other ways.

I hope it makes it in in some way.  i guess the the linux desktop developer crowd themselves are partly to blame for the state of what it takes to write to the monitor. just so crazy!!!

**Comment by justyntemme** [comment]
Due to recent events I have a lot more time on my hands, and would still be happy as ever to take direction and implement.
